webapp/javascript/components/TagsBar.tsx

```diff
@@ -102,6 +110,33 @@ function QueryInput({ initialQuery, onSubmit }: QueryInputProps) {
     }
   }, [query]);
 
+  useEffect(() => {
```

Comment:
In a high level can you explain what's goin on here?

Reply:
We have to keep `<code>` width and height exactly as `<textarea>` width and height. This effect allows us to recalculate them after updating any of useEffect dependencies. We get width/height from `textareaRef?.current?.['offsetWidth' | 'offsetHeight']`, this is the most accurate way to get dimensions of `<textarea>`

Reply:
There are 2 approaches for changing textarea height: changing height directly <textarea style={{height: "100px"}}>, and changing amount of rows <textarea rows={3}>. In this case changing rows in more convenient than changing textarea height (at least it is less demanding). onChange consists logic of incrementing or decrementing amout of textarea's rows depending on amount of symbols

Reply:
I see! I'm just not a big fan of calling `e.target` directly as opposed to going through the whole react cycle (use a state, then that using that state to set the `rows/scrollTop` / or using `refs`). I don't have any strong arguments except "it looks too imperative/I don't see many docs using that approach". Any thoughts on that?

That being said, can you explain a similar explanation you gave above above the `const onChange` line? I come from the principle I (Eduardo) got slightly confused by that, then maybe others will also will.

Reply:
My fault, your comment made me to go to the `react-textarea-autosize` src code to check what is going on under the hood, and it looks like we can get rid of these uncontrolled operations on Pyroscope side. Actions with `event` duplicated logic there. Sorry, I haven't checked it before I changed `<textarea>` to `TextareaAutosize`.

Reply:
No worries that's what reviews are for :P